### ISSUE TITLE
false hint for unconditionally true or false

### ISSUE THREAD

**Issue by zoechi** [dart-lang/linter#576]
> INFO: Conditions should not unconditionally evaluate to "TRUE" or to "FALSE" ([my_proj] lib/components/virtual_scroll/virtual_scroll_service.dart:55)

```dart
  int _rowsPerPage = 4;
  int _itemsPerRow = 5;
  int _itemHeight = 150;

    if ((_rowsPerPage ?? 0) <= 0 ||
        (_itemsPerRow ?? 0) <= 0 ||
        (_itemHeight ?? 0) <= 0) {
      return;
    }
```
for the whole `if() {}` block

**Comment by alexeieleusis** [comment]
We are not handling correctly `??`. I will fix it.

**Comment by alexeieleusis** [comment]
You are right @zoechi  this is the same than #658, please reopen if I am wrong.
